File: time/detail/iso.week.test.gen.php
<?php

    // Copyright (c) 2022 Mikael Simonsson <https://mikaelsimonsson.com>.
    // SPDX-License-Identifier: BSL-1.0

    if (!date_default_timezone_set('UTC'))
    {
        echo "Failed to set UTC timezone.\n";
        exit(1);
    }

    function gen($year, $month, $day)
    {
        $timestamp = gmmktime(0, 0, 0, $month, $day, $year);

        echo 'snn_require(time::point{', $year, ', ', $month, ', ', $day, '}.week()';
        echo ' == time::year_week{';
        echo date('o', $timestamp), ', ', ltrim(date('W', $timestamp), '0');
        echo '});', "\n";
    }

    for ($year = 1990; $year < 2030; ++$year)
    {
        for ($day = 1; $day <= 4; ++$day)
        {
            gen($year, 1, $day);
        }

        for ($day = 28; $day <= 31; ++$day)
        {
            gen($year, 12, $day);
        }
    }

File: time/detail/iso.week.test.gen.go
// Copyright (c) 2022 Mikael Simonsson <https://mikaelsimonsson.com>.
// SPDX-License-Identifier: BSL-1.0

package main

import "fmt"
import "time"

func main() {
	for year := 1990; year < 2030; year++ {
		for day := 1; day <= 4; day++ {
			y, w := time.Date(year, time.January, day, 0, 0, 0, 0, time.UTC).ISOWeek()
			fmt.Printf("snn_require(time::point{%d, 1, %d}.week()", year, day)
			fmt.Printf(" == time::year_week{%d, %d});\n", y, w)
		}

		for day := 28; day <= 31; day++ {
			y, w := time.Date(year, time.December, day, 0, 0, 0, 0, time.UTC).ISOWeek()
			fmt.Printf("snn_require(time::point{%d, 12, %d}.week()", year, day)
			fmt.Printf(" == time::year_week{%d, %d});\n", y, w)
		}
	}
}
